src/general/IWorldEffect.cs

```diff
@@ -37,13 +38,24 @@ public void OnTimePassed(double elapsed, double totalTimePassed)
         {
             var patch = targetWorld.Map.Patches[key];
 
-            foreach (var compound in patch.Biome.Compounds.Keys)
+            // If there are microbes to be eating up the primordial soup, reduce the milk
+            if (patch.SpeciesInPatch.Keys.Count > 0)
             {
-                if (compound.InternalName == "glucose")
+                Compound glucose = null;
+
+                foreach (var compound in patch.Biome.Compounds.Keys)
                 {
-                    var data = patch.Biome.Compounds[compound];
+                    if (compound.InternalName == "glucose")
+                    {
+                        glucose = compound;
+                    }
+                }
 
-                    data.Density *= Constants.GLUCOSE_REDUCTION_RATE;
+                if (glucose != null)
```

Comment:
This whole part could be replaced with code like: `if(pathc.Biome.Compounds.TryToGetValue(glucose, out ...))`
if you add a around line 36 (so it isn't done again every loop iteration): `var glucose = SimulationParameters.Instance.GetCompound("glucose");`